openslides_backend/action/agenda_item/update.py

```diff
@@ -22,3 +26,67 @@ class AgendaItemUpdate(UpdateAction):
             "duration",
         ]
     )
+
+    def calc_is_internal(self, type_: int, parent_is_internal: Optional[bool]) -> bool:
+        return type_ == AgendaItem.INTERNAL_ITEM or bool(parent_is_internal)
+
+    def calc_is_hidden(self, type_: int, parent_is_hidden: Optional[bool]) -> bool:
+        return type_ == AgendaItem.HIDDEN_ITEM or bool(parent_is_hidden)
+
+    def handle_children(
+        self, id_: int, parent_is_hidden: bool, parent_is_internal: bool
+    ) -> ActionPayload:
+        instances = []
+        agenda_item = self.datastore.get(
+            FullQualifiedId(self.model.collection, id_), ["child_ids"]
+        )
+        if agenda_item.get("child_ids"):
+            for child_id in agenda_item["child_ids"]:
+                child_ai = self.datastore.get(
+                    FullQualifiedId(self.model.collection, child_id), ["type"]
+                )
+                instance = dict()
+                instance["id"] = child_id
+                instance["is_hidden"] = self.calc_is_hidden(
+                    child_ai["type"], parent_is_hidden
+                )
+                instance["is_internal"] = self.calc_is_internal(
+                    child_ai["type"], parent_is_internal
+                )
```

Comment:
Please fetch `is_hidden` and `is_internal` as well and only update the instances where something was changed. If nothing was changed, you also don't have to handle the children, since they can't change, either.

Reply:
Done, only update instance and handle children if s.th. changed.